x/stakeibc/keeper/unbonding_records.go

```diff
@@ -43,171 +69,261 @@ func (k Keeper) CreateEpochUnbondingRecord(ctx sdk.Context, epochNumber uint64)
+		balancedDelegation := balancedDelegation[validator.Address]
+		capacity := validator.Delegation.Sub(balancedDelegation)
+		if capacity.IsPositive() {
+			validatorCapacities = append(validatorCapacities, ValidatorUnbondCapacity{
+				ValidatorAddress:   validator.Address,
+				Capacity:           capacity,
+				CurrentDelegation:  validator.Delegation,
+				BalancedDelegation: balancedDelegation,
+			})
+		}
 	}
 
-	// Check if each validator has enough current delegations to cover the target unbonded amount
-	// If it doesn't have enough, update the target to equal their total delegations and record the overflow amount
-	finalUnbondingsByValidator := make(map[string]sdkmath.Int)
-	overflowAmount := sdkmath.ZeroInt()
-	for _, validator := range hostZone.Validators {
-
-		targetUnbondAmount := targetUnbondingsByValidator[validator.Address]
-		k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId,
-			"  Validator %s - Weight: %d, Target Unbond Amount: %v, Current Delegations: %v", validator.Address, validator.Weight, targetUnbondAmount, validator.Delegation))
+	return validatorCapacities
+}
 
-		// If they don't have enough to cover the unbondings, set their target unbond amount to their current delegations and increment the overflow
-		if targetUnbondAmount.GT(validator.Delegation) {
-			overflowAmount = overflowAmount.Add(targetUnbondAmount).Sub(validator.Delegation)
-			targetUnbondAmount = validator.Delegation
+// Sort validators by the ratio of the ideal balanced delegation to their current delegation
+// This will sort the validator's by how proportionally unbalanced they are
+//
+// Ex:
+//   Val1: Ideal Balanced Delegation 80,  Current Delegation 100 (surplus of 20), Ratio: 0.8
+//   Val2: Ideal Balanced Delegation 480, Current Delegation 500 (surplus of 20), Ratio: 0.96
+// While both validators have the same net unbalanced delegation, Val2 is proportionally
+//    more balanced since the surplus is a smaller percentage of it's overall delegation
+//
+// This will also sort such that 0-weight validator's will come first as their
+//   ideal balanced delegation will always be 0, and thus their ratio will always be 0
+// If the ratio's are equal, the validator with the larger delegation/capacity will come first
+func SortUnbondingCapacityByPriority(validatorUnbondCapacity []ValidatorUnbondCapacity) []ValidatorUnbondCapacity {
+	lessFunc := func(i, j int) bool {
+		validatorA := validatorUnbondCapacity[i]
+		validatorB := validatorUnbondCapacity[j]
+
+		// Sort by the balance ratio first - in ascending order - so the more unbalanced validators appear first
+		if !validatorA.GetBalanceRatio().Equal(validatorB.GetBalanceRatio()) {
+			return validatorA.GetBalanceRatio().LT(validatorB.GetBalanceRatio())
 		}
-		finalUnbondingsByValidator[validator.Address] = targetUnbondAmount
-	}
 
-	// If there was overflow (i.e. there was at least one validator without sufficient delegations to cover their unbondings)
```

Comment:
I love how much simpler and more understandable the new code is.  Deleting lines is that much less surface area for bugs